### PR TITLE
Fix a typo in warning message inside Trainer.reset_train_dataloader

### PR DESCRIPTION
**Pull request by semaphore-egg**
## What does this PR do?

Fix a small typo in a warning message.

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Yes
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by akihironitta — APPROVED** [review]

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -1885,7 +1885,7 @@ def reset_train_dataloader(self, model: Optional["pl.LightningModule"] = None) -
 
         if self.loggers and self.num_training_batches < self.log_every_n_steps:
             rank_zero_warn(
-                f"The number of training samples ({self.num_training_batches}) is smaller than the logging interval"
+                f"The number of training batches ({self.num_training_batches}) is smaller than the logging interval"
```
Comment (awaelchli) [inline pytorch_lightning/trainer/trainer.py:1888]: I recall having added a test for this, I hope this did not get lost. Can you check if it's there and update it with the change here?

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/trainer/trainer.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/trainer/trainer.py:1888]: Oh, found it. I made the change directly to your PR. Sorry, I initially thought the PR was by a staff-contributor, hence my direct request.

**Comment by awaelchli** [comment]
Thanks for the fix!

**Review by Borda — APPROVED** [review]

**Comment by semaphore-egg** [comment]
> Thanks for the fix!

Haha, feel really excited that my first PR in Github was approved.

**Pull request merged**